atsign-foundation/at_server
# fix: client commit log compaction changes

**- What I did**
- The design changes for the compaction job
- Added generics for AtCommitLog and AtAccessLog and for the methods inside the classes
- Added unit tests around compaction job and writing stats to the keystore

**- How I did it**
- Introduced new abstraction AtCompaction. CommitLog, AccessLog and NotificationKeystore will implement this new 
  abstraction
- Deprecated TimeBasedCompaction and SizeBasedCompaction and deleted the files.
- Modified implementation in AtCompactionService

**- How to verify it**
 -  Unit tests in at_persistence_secondary_server to verify the compaction job changes
 -  In at_secondary_server, pubspec.yaml, override the at_persistence_secondary_server to point to "this" branch to ensure the 
  changes do not break the server code.
